.changeset/healthy-cougars-relax.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+Add build concurrency config option
```

Comment:
The `buildConfig`, which is where you are setting this, is not part of the config object. It's something that can be modified via an integration. For example you could do this:

```js
export default defineConfig({
  integrations: [
    {
      name: 'my-integration',
      hooks: {
        'astro:build:start': ({ buildConfig }) {
          buildConfig.concurrency = 8;
        }
      }
    }
  ]
});
```

We've never discussed making the `buildConfig` be exposed by the top-level config object. I think that might require an RFC if it led to nesting, which we don't currently have. I think leaving this as an advanced API that you can add via an integration is a better way to go, for now.

Reply:
Sure, can be changed to use `opts.astroConfig` and move `concurrency` from `build` to the root of `AstroConfigSchema` then?

In that case, it would be defined in:
```
export default defineConfig({
	// integrations: [...],
	concurrency: 8,
	// adapter: ...
});
```

We could also rename the key in that case if you prefer.

Reply:
We can move it to the top-level config, but that's going to take some convincing with the core team. I'll bring it up to them.

Reply:
I'm a light +1 on a top-level `build.concurrency` astro config value, and definitely prefer it to the hidden integration-only `buildConfig` object